python_modules/dagster/dagster/_core/definitions/asset_check_execution.py

```diff
@@ -0,0 +1,116 @@
+import enum
+from typing import TYPE_CHECKING, NamedTuple, Optional
+
+import dagster._check as check
+from dagster._serdes.serdes import whitelist_for_serdes
+
+from .asset_check_result import AssetCheckResult
+from .events import AssetKey
+
+if TYPE_CHECKING:
+    from dagster._core.event_api import EventLogRecord
+
+
+@whitelist_for_serdes
+class AssetCheckExecutionStatus(enum.Enum):
+    PLANNED = "PLANNED"  # No result yet
+    RESULT_SUCCESS = "RESULT_SUCCESS"
+    RESULT_FAILURE = "RESULT_FAILURE"
+
+
+@whitelist_for_serdes
+class AssetCheckExecution(
+    NamedTuple(
+        "_AssetCheckExecution",
+        [
+            ("asset_key", AssetKey),
+            ("check_name", str),
+            ("check_status", AssetCheckExecutionStatus),
+            ("run_id", str),
+            ("check_result", Optional[AssetCheckResult]),
+            ("target_materialization_record", Optional["EventLogRecord"]),
```

Comment:
Why store the materialization record instead of just a reference to it (storage_id)?